### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by Spuddy** [tim-sueberkrueb/switch#2]
Hello, 
Great app :+1: , a couple of ideas after about 20mins of playing it, take them as you wish: 

1) When moving slider to multiply, it'd be nice to be able to tap the plus or minus sign to increment by 1, which would be easier than sliding it. 
2) It might be nice to see the final matrix at the end before it shows the success screen, so you can visualise what the final result was. 
3) An option to do divide manually rather than having it pop up on the side of the matrix, perhaps a toggle between multiply and divide, not sure how you'd stop people from making ugly decimals (display as fractions maybe, still could get messy) 
4) Custom matrices, ability to put custom numbers in to solve for a problem using this as a nifty tool.

**Comment by timsueberkrueb** [comment]
Hey @Spuddy, thanks for the suggestions. 

You find my thoughts below:

> 1) When moving slider to multiply, it'd be nice to be able to tap the plus or minus sign to increment by 1, which would be easier than sliding it. 

In an earlier version, there where multiply fields like this:
![Multply fields](https://cloud.githubusercontent.com/assets/11774614/18027210/5685a56c-6c5e-11e6-9aca-d7882d519439.png)
... but it took ages to multiply. The slider is much quicker. 
Another problem is that I can hardly imagine that it would work with both sliding and incrementing by clicking. E.g. how to accept the right value (releasing the slider)? It's would also lead to confusing if there where two ways to do it.

>  2) It might be nice to see the final matrix at the end before it shows the success screen, so you can visualise what the final result was. 

I agree!

>  3) An option to do divide manually rather than having it pop up on the side of the matrix, perhaps a toggle between multiply and divide, not sure how you'd stop people from making ugly decimals (display as fractions maybe, still could get messy) 

What would be the advantage of letting the user multiply manually? Switch automatically detects the greatest common divisor for you. I also would like to avoid fractions to keep it simple :)

>  4) Custom matrices, ability to put custom numbers in to solve for a problem using this as a nifty tool. 

Well, yeah, good idea. Though that might be better in a separate app (which also can automatically solve them for you)? Switch is more like an (educational) game rather than a tool.

**Comment by Spuddy** [comment]
Hi @tim-sueberkrueb, I see you're points, it makes sense. 
One of the best maths-games I've played yet :smile:

**Comment by timsueberkrueb** [comment]
@Spuddy reopening this, I'll at least implement 2. :)

**Comment by timsueberkrueb** [comment]
Small update, I created an entirely new version of Switch where I also implemented 2. :)
I'm not sure when I will find the time to push the repo and finish it though. Also, it doesn't support Ubuntu Phone right now (as Ubuntu Phone uses an older Qt version right now).
It will probably support Ubuntu Phones as soon as they switch to (or rather new phones come with) snap.
But for the current Ubuntu version, this won't fix, so I'm closing this issue.
